src/gen_rules.ml

```diff
@@ -464,19 +464,18 @@ module Gen(P : Params) = struct
         libs @ List.map ~f:(Path.change_extension ~ext:ctx.ext_obj) cm
     in
     SC.add_rule sctx
-      ((libs_and_cm >>> Build.dyn_paths (Build.arr objs))
-       &&&
-       Build.fanout
-       (Ocaml_flags.get flags mode)
-       (SC.expand_and_eval_set sctx ~scope ~dir link_flags ~standard:[])
+      (Build.fanout3
+         (libs_and_cm >>> Build.dyn_paths (Build.arr objs))
+         (Ocaml_flags.get flags mode)
+         (SC.expand_and_eval_set sctx ~scope ~dir link_flags ~standard:[])
```

Comment:
See original review comment on this at https://github.com/janestreet/jbuilder/pull/153/files#r131172546

Really, this should use Build.fanout4 (which was subsequently added during that PR, IIRC)

Reply:
Hmm, I gave it a try but I'm not sure how to do it with Build.fanout4 with the jsoo rules complicating things. If you know how, please push a patch here.